lib/rails_admin/abstract_model.rb

```diff
@@ -87,12 +87,11 @@ def each_associated_children(object)
         case association.type
         when :has_one
           if child = object.send(association.name)
-            yield(association, child)
+            yield(association, [child])
           end
         when :has_many
-          object.send(association.name).each do |child| # rubocop:disable ShadowingOuterLocalVariable
-            yield(association, child)
-          end
+          children = object.send(association.name)
+          yield(association, Array.new(children))
```

Comment:
I don't think that children are fetched from database here.
`children` variable contains a `CollectionProxy` returned to `_delete_notice.html.haml` where `count` and `'first(limit)` is called. First n children are fetched here.
Am I wrong ?